src/App.js

```diff
@@ -8,14 +8,16 @@ import HomePage from "./pages/HomePage";
 
 function App() {
   return (
-    <div className="App">
+    <div className='App'>
       <NavBar />
-      <Switch>
-        <Route path="/discover/:searchtext?" component={DiscoverMoviesPage} />
-        <Route path="/movie/:imdb_id" component={MoviePage} />
-        <Route path="/about" component={AboutPage} />
-        <Route path="/" component={HomePage} />
-      </Switch>
+      <Routes>
+        <Route path='/discover' element={<DiscoverMoviesPage />}>
+          <Route path=':searchtext' element={<DiscoverMoviesPage />} />
+        </Route>
```

Comment:
The conditional route param `/:searchtext?` is not a thing anymore. We now have to nest additions to the url and match to the same (or different) component

Reply:
Also use `element` instead of `component`